### PR TITLE
Registers sin function

### PR DESCRIPTION
**Pull request by shivimathur**
Towards : https://github.com/lfortran/lfortran/issues/3416

### CODE REVIEW

**Comment by Pranavchiku** [comment]
@shivimathur let's also add a simple test, as given in #3416.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -656,6 +656,8 @@ RUN(NAME intrinsics_144 LABELS gfortran llvm) # dot_product
 RUN(NAME intrinsics_145 LABELS gfortran llvm) # dot_product
 RUN(NAME intrinsics_146 LABELS gfortran llvm) # dprod
 
+RUN(NAME register_intrinsic_sin LABELS gfortran llvm)
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:659]:
```suggestion
RUN(NAME register_intrinsic_sin LABELS gfortran llvm fortran)
```

**Review by Pranavchiku — APPROVED** [review]
Looks good, added a comment. Thank you!

File: integration_tests/register_intrinsic_sin.f90 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+program main
+real :: x
+print *, sin(x)
+end program
```
Comment (certik) [inline integration_tests/register_intrinsic_sin.f90:4]: I think this is not a good test:

* `x` is not initialized
* It needs to test that the result is correct (`if (abs(sin(x) /= ...) ... error stop`)
* The most important: we already have tests for `sin`. So this test is redundant. We should not be adding a new file like this for all intrinsics (there are dozens of them). Rather, let's take an existing test, and add a `fortran` flag to it. Perhaps more intrinsics must be fixed, but then we should do that.

**Review by certik — CHANGES_REQUESTED** [review]
Thanks for the PR and the fix!

I think this PR needs more work, I think it will add to our technical debt if we merge it as is. See my comment above.

File: integration_tests/register_intrinsic_sin.f90 (hunk 2 above)
Comment (Pranavchiku) [reply integration_tests/register_intrinsic_sin.f90:4]: Yes, I agree. I missed these, thanks for reviewing this :)

**Comment by shivimathur** [comment]
Thanks for the review. I have made the changes. This is ready to review again.

**Comment by certik** [comment]
Tests fail, so they need to be fixed first.

**Comment by shivimathur** [comment]
This is now ready for review?

**Review by certik — APPROVED** [review]
I think this is fine, thanks!

**Pull request merged**